### PR TITLE
Feature/jsonify

### PR DESCRIPTION
**Pull request by southeo**
Jsonify Request bodies and response objects for mapping and source system endpoints.

### CODE REVIEW

File: src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java [hunk 1]
```diff
@@ -25,45 +31,53 @@
 
 @Slf4j
 @RestController
-@RequestMapping("/source-system")
+@RequestMapping("orchestrator/source-system")
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java:34]: Why ochestrator? I don't think this is necessary as we can put that in the route path and let the application work from there.
The path will then be `/api/v1/orchestration/source-system` (for example) and the application only receives `/source-system`

File: src/main/java/eu/dissco/orchestration/backend/domain/HandleType.java [hunk 2]
```diff
@@ -1,5 +1,18 @@
 package eu.dissco.orchestration.backend.domain;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 public enum HandleType {
-  MAPPING, SOURCE_SYSTEM;
+  @JsonProperty("mapping")MAPPING("mapping"), @JsonProperty("sourceSystem")SOURCE_SYSTEM("sourceSystem");
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/domain/HandleType.java:6]: Oh could this be formatted different.
Each enum on a new line

File: src/main/java/eu/dissco/orchestration/backend/service/MappingService.java [hunk 3]
```diff
@@ -24,41 +25,50 @@ public class MappingService {
...
+  }
+
+  private JsonApiWrapper wrapSingleResponse(String id, MappingRecord mappingRecord, String path){
+    return new JsonApiWrapper(
+        new JsonApiData(id, HandleType.MAPPING, mapper.valueToTree(mappingRecord.mapping())),
+        new JsonApiLinks(path)
+    );
+  }
+
+  public JsonApiListWrapper getMappings(int pageNum, int pageSize, String//When
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/service/MappingService.java:58]: There is a `//When` comment

File: src/test/java/eu/dissco/orchestration/backend/testutils/TestUtils.java [hunk 4]
```diff
@@ -24,13 +27,53 @@ public class TestUtils {
   public static final String OBJECT_DESCRIPTION = "Source system for the DWCA of the Tunicate specimen";
   public static final Instant CREATED = Instant.parse("2022-11-01T09:59:24.00Z");
   public static final ObjectMapper MAPPER = new ObjectMapper().findAndRegisterModules();
-  public static final String SANDBOX_URI = "https://sandbox.dissco.tech/";
+  public static final String SANDBOX_URI = "https://sandbox.dissco.tech/orchestrator";
```
Comment (samleeflang) [inline src/test/java/eu/dissco/orchestration/backend/testutils/TestUtils.java:30]: Think this is an issue as sourcesystem has /orchestrator/source-system but mapping is just /mapping

**Review by samleeflang — COMMENTED** [review]
As we discussed this afternoon.
A new user story is needed to make sure this complies to the JSON:API specification.
Read that carefully and make sure you got all the checks and requirements.

File: src/test/java/eu/dissco/orchestration/backend/testutils/TestUtils.java (hunk 4 above)
Comment (southeo) [reply src/test/java/eu/dissco/orchestration/backend/testutils/TestUtils.java:30]: You're right, they should both be `/orchestrator`. Fixed

**Comment by southeo** [comment]
To fully comply with JsonApi, we need to allow the user to provide incomplete records (i.e. records with only a first name to be changed) on PATCH requests. [User Story](https://naturalis.atlassian.net/browse/DD-364?atlOrigin=eyJpIjoiZDA2NTIyN2E0YTY4NDNjMmI3N2Q0YmJkOTlmOGExOTIiLCJwIjoiaiJ9)

**Pull request merged**